binaryen-sys/Shim.cpp

```diff
@@ -56,3 +56,9 @@ extern "C" void BinaryenShimDisposeBinaryenModuleAllocateAndWriteResult(
         free(result.sourceMap);
     }
 }
+
+// NOTE: this is highly unoptimal, but this is the only API available
+extern "C" bool BinaryenShimPassFound(const char *pass) {
```

Comment:
The other way we **could** solve this, is by listing all available passes in an enum. Then `run_optimization_passes` takes a slice of these enum values and calls an internal method on each of the enum values to get the corresponding name of the pass, maybe already in the form of a pointer to a C const string.

In this scheme we will not need `BinaryenShimPassFound` shim. We will have to synchronize binaryen-sys with binaryen more closely though.

Reply:
> The other way we could solve this, is by listing all available passes in an enum.

That was my goal later to have a Rust data structure for the passes, usable by the caller to see what is available as well as by the runner, see the message in the opening description:

> Part 2 adds an interface to retrieve the available pass names.

Reply:
Changed the other two methods, but would prefer to postpone this bigger change (will ask for more help in it).